### PR TITLE
763 side navigation layout guidance page

### PR DESCRIPTION
**Pull request by GCHQ-Developer-112**
## Summary of the changes

- Add navigation patterns guidance
- Add top navigation and side navigation pattern code examples for React and web components, compatible with Javascript
- Add stackblitz button functionality along with helper functions

## Related issue
#760 #763 https://github.com/mi6/ic-design-system/issues/764 https://github.com/mi6/ic-design-system/issues/765 https://github.com/mi6/ic-design-system/issues/758 #751 

## Checklist

- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

**Comment by GCHQ-Developer-112** [comment]
[Branch deployment](https://animated-fiesta-gvyom1k.pages.github.io/branches/763-side-navigation-layout-guidance-page/patterns)

**Comment by gd2910** [comment]
I'm getting some  weird navigation-item heights when expanded
image
Fixed by getting rid of the calculated height, but ~annoyingly~ interestingly when in collapsed satte the nav-items have the same issue as above.
image

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]
Looks amazing 😃 Noticed a few things but only very small

I can't view the StackBlitz examples at the moment because of the issues with builds loading properly (I think the StackBlitz buttons can only appear on the build and not locally?)

**Comment by GCHQ-Developer-112** [comment]
> Looks amazing 😃 Noticed a few things but only very small
> 
> I can't view the StackBlitz examples at the moment because of the issues with builds loading properly (I think the StackBlitz buttons can only appear on the build and not locally?)

Yes, on line 216, the showStackblitzBtn prop is dependent on whether the environment is prod or dev >> `showStackblitzBtn={!!process.env.GATSBY_GA_TRACKING_ID && type === "pattern"}`
 
 To see the Stackblitz button in your local environment, you'll need to remove `!!process.env.GATSBY_GA_TRACKING_ID && `  so the prop is `showStackblitzBtn={type === "pattern"}` instead

File: src/icds-pages-data.json
[the remark this answers is not in the document]
Comment (GCHQ-Developer-112) [reply src/icds-pages-data.json:1]: As there's some new guidance page changes, I committed the regenerated docs file. I'm not aware of a way to stop the dates from being changed whilst keeping the new guidance in the docs, without going through the docs and reverting each date changed

Does the date change have a big impact?

File: src/components/CodePreview/index.tsx [hunk 1]
```diff
@@ -1,66 +1,143 @@
...
-        <SlottedSVG path={mdiContentCopy} slot="left-icon" />
-      </ic-button>
-    </div>
-  </>
-);
+      <div className={clsx(type === "pattern" && type, "snippet-container")}>
+        {type === "pattern" && (
+          <ic-button
+            variant="tertiary"
+            size={isLargeViewport ? "small" : "default"}
```
[the remark this answers is not in the document]
Comment (GCHQ-Developer-112) [reply src/components/CodePreview/index.tsx:92]: No, this is the correct way round. At the moment, the default height of the icon button (32px) is not the same as the default height of the regular button (40px). To get to be the same height, I need to set the size of the regular button to `small`

**Comment by GCHQ-Developer-112** [comment]
> Just taking a look at the StackBlitz demo;
> 
> I'm getting some weird navigation-item heights when expanded image Fixed by getting rid of the calculated height, but ~annoyingly~ interestingly when in collapsed satte the nav-items have the same issue as above. image

I've now created a PR to fix this issue: [Add auto height to `ic-navigation-item` when it's in the side menu to prevent it stretching when used in StackBlitz](https://github.com/mi6/ic-ui-kit/pull/1745)

File: src/icds-pages-data.json
[the remark this answers is not in the document]
Comment (GCHQ-Developer-847) [reply src/icds-pages-data.json:1]: Ahh sorry - I was thinking these dates corresponded to the "last reviewed" ones at the bottom of the page, ignore me! All good and makes sense 👍

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**